### PR TITLE
Bso smithing boosts

### LINKED ISSUE #4146: Smithing master cape, dwarven gauntlets and blacksmith armour isn't working at blast furnace.
How it used to be
image

How it is currently
image

### PR DESCRIPTION
**Pull request by TastyPumPum**
As per the wiki, added the blacksmith equipment 10% bonus xp, along with the smithing master cape 2x boost to blast furnance, which I believe got missed during slash migration.
https://bso-wiki.oldschool.gg/skills/smithing

Closes #4146 

-   [x] I have tested all my changes thoroughly.

### CODE REVIEW

**Review by themrrobert — CHANGES_REQUESTED** [review]

**Comment by gc** [comment]
>which I believe got missed during slash migration.

should also verify this

File: src/tasks/minions/smeltingActivity.ts [hunk 1]
```diff
@@ -51,7 +54,13 @@ export default class extends Task {
 		}
 
 		if (masterCapeInEffect) {
-			str += '\n2x less likely to fail from Smithing master cape';
+			if (!(blastf && bar.id === itemID('Iron bar'))) {
```
[the remark this answers is not in the document]
Comment (TastyPumPum) [reply src/tasks/minions/smeltingActivity.ts:57]: Just tested again, it shows this message when smelting dwarven bars through regular smelting. Unable to smelt them through blast furnace.

**Comment by TastyPumPum** [comment]
> > which I believe got missed during slash migration.
> 
> should also verify this

Not quite sure how you mean but I have linked below various times the boosts can be seen in the code. Also don't see why the wiki would state it if it was never in the code, but can understand the caution on adding big boosts. 

Dwarven gauntlets in the smelting code can be seen here - https://github.com/imgidedin/GidOSBot/blob/48c772124ecf4b387088622f1efde500b2c8213b/src/commands/Minion/smelt.ts
Blacksmith outfit can be seen here - https://github.com/oldschoolgg/oldschoolbot/pull/2660
Smithing mastercape perk can be seen here - https://github.com/imgidedin/GidOSBot/blob/48c772124ecf4b387088622f1efde500b2c8213b/src/commands/Minion/blastfurnace.ts

Discord message around the time of moving stuff to blast furnace saying not to worry about blast furnace for bso - https://discordapp.com/channels/342983479501389826/648196527294251020/972119017575833650

So that's probably why it got missed, hope all this is sufficient but can try and find more if you need.

File: src/mahoji/commands/smelt.ts [hunk 2]
```diff
@@ -120,8 +117,25 @@ export const smeltingCommand: OSBMahojiCommand = {
...
+				timeToSmithSingleBar /= 2;
+				boosts.push('2x boost for Smithing master cape');
+			}
+			if (user.hasItemEquippedAnywhere(itemID('Dwarven gauntlets')) && bar.id !== itemID('Gold bar')) {
+				boosts.push('2x boost for having a Dwarven gauntlets equipped.');
+				timeToSmithSingleBar /= 2;
+			}
 		}
 
+		if (!blast_furnace) {
```
Comment (themrrobert) [inline src/mahoji/commands/smelt.ts:130]: This is incorrect. 
In revision b7c0e151 (blastfurnace.ts) you can see that blast furnace DOES get 2x boost for Dwarven gauntlets EXCEPT for when smithing Gold.

File: src/mahoji/commands/smelt.ts [hunk 3]
```diff
@@ -120,8 +117,25 @@ export const smeltingCommand: OSBMahojiCommand = {
 				timeToSmithSingleBar *= 1.075;
 				boosts.push('-7.5% penalty for not having graceful equipped.');
 			}
+			if (user.hasItemEquippedAnywhere('Smithing master cape')) {
+				timeToSmithSingleBar /= 2;
+				boosts.push('2x boost for Smithing master cape');
+			}
+			if (user.hasItemEquippedAnywhere(itemID('Dwarven gauntlets')) && bar.id !== itemID('Gold bar')) {
```
Comment (themrrobert) [inline src/mahoji/commands/smelt.ts:124]: This is wrong also. Dwarven gauntlets are supposed to give Gold the 2x boost, except when blast furnace. (smelt.ts rev: b7c0e151 )

**Review by themrrobert — CHANGES_REQUESTED** [review]
This is based on faulty assumptions, please review revision b7c0e151 in the referenced sections so this can be accurate, thanks :)

File: src/mahoji/commands/smelt.ts (hunk 3 above)
Comment (TastyPumPum) [reply src/mahoji/commands/smelt.ts:124]: Reviewed the blastfurnace.ts file and this looks to match, this code is under the header if (blast_furnace) so looks and tests good to me.

File: src/mahoji/commands/smelt.ts (hunk 2 above)
Comment (TastyPumPum) [reply src/mahoji/commands/smelt.ts:130]: This implies everywhere thats not blast_furnance gets the boost, so this with the previous one should cover everything getting the boost except blast furnancing gold

**Comment by TastyPumPum** [comment]
Reopened as I have further tested this and belive it is working as per rev: https://github.com/oldschoolgg/oldschoolbot/commit/b7c0e151fe73b3bcac13676833eb2c1a4142230d

File: src/mahoji/commands/smelt.ts [hunk 4]
```diff
@@ -120,8 +117,25 @@ export const smeltingCommand: OSBMahojiCommand = {
...
+				boosts.push('2x boost for having a Dwarven gauntlets equipped.');
+				timeToSmithSingleBar /= 2;
+			}
 		}
 
+		if (!blast_furnace) {
+			if (user.hasItemEquippedAnywhere(itemID('Dwarven gauntlets'))) {
+				boosts.push('2x boost for having a Dwarven gauntlets equipped.');
```
Comment (gc) [inline src/mahoji/commands/smelt.ts:132]: shouldnt be 'a' here

File: src/mahoji/commands/smelt.ts [hunk 5]
```diff
@@ -120,8 +117,25 @@ export const smeltingCommand: OSBMahojiCommand = {
...
+				boosts.push('2x boost for Smithing master cape');
+			}
+			if (user.hasItemEquippedAnywhere(itemID('Dwarven gauntlets')) && bar.id !== itemID('Gold bar')) {
+				boosts.push('2x boost for having a Dwarven gauntlets equipped.');
+				timeToSmithSingleBar /= 2;
+			}
 		}
 
+		if (!blast_furnace) {
+			if (user.hasItemEquippedAnywhere(itemID('Dwarven gauntlets'))) {
```
Comment (gc) [inline src/mahoji/commands/smelt.ts:131]: no need for itemID here (And elsewhere in this pr)

**Review by gc — CHANGES_REQUESTED** [review]

File: src/mahoji/commands/smelt.ts (hunk 5 above)
Comment (TastyPumPum) [reply src/mahoji/commands/smelt.ts]: Removed itemID from this PR

Comment (TastyPumPum) [reply src/mahoji/commands/smelt.ts]: Think I misinterpreted this, so added itemID back in some situations, let me know if you need it removed elsewhere

**Comment by TastyPumPum** [comment]
Any further changes needed on this? Works how it should in testing, the same as it was pre mahoji changes.

**Pull request merged**